src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                                        fmt::format("Invalid curve name for EIR FFHP (name={}; entered curve name: {}",
+                                                    thisPLHP.name,
+                                                    auxEIRFTName.get<std::string>()));
+                        errorsFound = true;
+                    }
+                } else {
+                    thisPLHP.auxElecEIRFoTempCurveIndex = 0;
+                }
+
+                // A18 auxiliary_electric_energy_input_ratio_function_of_plr_curve_name
+                if (fields.find("auxiliary_electric_energy_input_ratio_function_of_plr_curve_name") != fields.end()) {
+                    auto &auxEIRFPLRName = fields.at("auxiliary_electric_energy_input_ratio_function_of_plr_curve_name");
+                    thisPLHP.auxElecEIRFoPLRCurveIndex =
+                        Curve::GetCurveIndex(state, UtilityRoutines::MakeUPPERCase(auxEIRFPLRName.get<std::string>()));
+                    if (thisPLHP.auxElecEIRFoPLRCurveIndex == 0) {
+                        ShowSevereError(state,
+                                        fmt::format("Invalid curve name for EIR FFHP (name={}; entered curve name: {}",
+                                                    thisPLHP.name,
+                                                    auxEIRFPLRName.get<std::string>()));
+                        errorsFound = true;
+                    }
+                } else {
+                    thisPLHP.auxElecEIRFoPLRCurveIndex = 0;
+                }
+
+                // N12 standby_electric_power
+                if (fields.find("standby_electric_power") != fields.end()) {
+                    thisPLHP.standbyElecPower = fields.at("standby_electric_power").get<Real64>();
+                } else {
+                    Real64 defaultVal = 0.0;
+                    if (!state.dataInputProcessing->inputProcessor->getDefaultValue(
+                            state, cCurrentModuleObject, "standby_electric_power", defaultVal)) {
+                        // this error condition would mean that someone broke the input dictionary, not their
+                        // input file.  I can't really unit test it so I'll leave it here as a severe error
+                        // but excluding it from coverage
+                        ShowSevereError(state,                                                                            // LCOV_EXCL_LINE
+                                        "EIR FFHP: standby electric power not entered and could not get default value."); // LCOV_EXCL_LINE
+                        errorsFound = true;                                                                               // LCOV_EXCL_LINE
+                    } else {
+                        thisPLHP.standbyElecPower = defaultVal;
+                    }
+                }
+
+                bool nodeErrorsFound = false;
+                thisPLHP.loadSideNodes.inlet = NodeInputManager::GetOnlySingleNode(state,
```

Comment:
📌 (i.e., not for this PR). One of my favorite EnergyPlus patterns. `GetOnlySingleNode` followed immediately and predictably by `GetOnlySingleNode`.

Reply:
> 📌 (i.e., not for this PR). One of my favorite EnergyPlus patterns. `GetOnlySingleNode` followed immediately and predictably by `GetOnlySingleNode`.

What's the concern here? Any given object could have one or more node names as inputs and typically at least one of the other arguments will be different.

Reply:
The concern is that the two calls to `GetOnlySingleNode` do a fair bit of redundant work. It would be better to have a function `getNodePair` that gets both nodes in one shot.

Reply:
I think I got the idea. That pair registering function might be good to simplify the general equipment nodes registering process with an inlet and outlet. It should work in many general scenarios, although it is still not 100% if sure this would be apply to any type of equipment (essentially this is assuming that there is only one-parameter difference between the inlet and the outlet---maybe true though). Since this would be a general change that would potentially applied to a wide range of equipment, it looks like it might be easier to address and test this idea in a separate PR.

Reply:
Oh, it was definitely intended for a separate PR (if any). That's what the pushpin (📌 ) emoji means, i.e., "put a pin in this for later".